app/services/ai_backend/anthropic.rb

```diff
@@ -28,6 +28,18 @@ def get_next_chat_message(&chunk_received_handler)
 
     response_handler = proc do |intermediate_response, bytesize|
       chunk = intermediate_response.dig("delta", "text")
+
+      if (input_tokens = intermediate_response.dig("message", "usage", "input_tokens"))
+        @message.input_token_count += input_tokens
+      end
+      # the initial number of output tokens is first returned in message.usage.output_tokens and
+      # the rest in usage.output_tokens in the final response
+      output_tokens = intermediate_response.dig("message", "usage", "output_tokens")
+      output_tokens ||= intermediate_response.dig("usage", "output_tokens")
+      if output_tokens
+        @message.output_token_count += output_tokens
+      end
```

Comment:
I checked with the anthropic SDK team and we only need the final output_tokens count. That number is cumulative. We can ignore the initial one: https://github.com/anthropics/anthropic-sdk-python/issues/424#issuecomment-2227894093

```suggestion
      if output_tokens = intermediate_response.dig("message", "usage", "output_tokens")
        @message.output_token_count += output_tokens
      end
```